packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.ts

```diff
@@ -0,0 +1,157 @@
+import { existsSync as _existsSync } from 'fs';
+import { execa as _execa } from 'execa';
+import * as path from 'path';
+import {
+  type PackageManagerName,
+  type PackageManagers,
+} from './package_manager.js';
+import { logger } from '../logger.js';
+import { NpmPackageManagerController } from './npm_package_manager_controller.js';
+import { PnpmPackageManagerController } from './pnpm_package_manager_controller.js';
+import { YarnClassicPackageManagerController } from './yarn_classic_package_manager_controller.js';
+import { YarnModernPackageManagerController } from './yarn_modern_package_manager_controller.js';
+import {
+  NpmProjectInitializer,
+  PnpmProjectInitializer,
+  YarnClassicProjectInitializer,
+  YarnModernProjectInitializer,
+} from '../project-initializer/index.js';
+
+export type DependencyType = 'dev' | 'prod';
+
+/**
+ *
+ */
+export abstract class PackageManagerController {
```

Comment:
could this just be a type?

Reply:
In this case, I think `type` and `abstract class` are the same. Since @sobolk made [the suggestion](https://github.com/aws-amplify/amplify-backend/pull/793#discussion_r1423259577), I'm curious about what he thinks

Reply:
an abstract class should have some sort of shared implementation that makes sense for all subclasses. Since there's no impl here, I think a type is better

Reply:
We need some "abstraction", we don't export this from package so this can be either `type` or `abstract class`.
My suggestion is to keep addressing other feedback and make this call later when we know if there's any `protected` state or logic left in this class. If there's none we can convert this to `type`.

Reply:
This is looking more and more like it should be a type rather than an abstract class